pkg/registry/generic/etcd/etcd.go

```diff
@@ -459,16 +460,46 @@ func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, lis
 	if err != nil {
 		return nil, err
 	}
-	for _, item := range items {
-		accessor, err := meta.Accessor(item)
-		if err != nil {
-			return nil, err
-		}
-		if _, err := e.Delete(ctx, accessor.GetName(), options); err != nil {
-			return nil, err
-		}
+	// Spawn a number of goroutines, so that each one is supposed to delete
+	// at most maxElemsPerWorker elements and they are running in parallel.
+	expectedMaxElemsPerWorker := 1000
+	workersNumber := (len(items) + expectedMaxElemsPerWorker - 1) / expectedMaxElemsPerWorker
+	wg := sync.WaitGroup{}
+	mu := sync.Mutex{}
+	currentElem := 0
+	wg.Add(workersNumber)
+	errs := make(chan error, workersNumber)
+	for i := 0; i < workersNumber; i++ {
+		go func() {
+			defer wg.Done()
+			for {
+				mu.Lock()
```

Comment:
rather than contending on a shared lock, would it make more sense to partition the list of items to delete and hand them to workers?